File: java/src/main/java/io/hron/ast/YearRepeat.java
package io.hron.ast;

import java.util.List;

/**
 * A year-based repeat expression like "every year on dec 25 at 00:00".
 *
 * @param interval the number of years between occurrences (1 for every year)
 * @param target the day within the year to fire
 * @param times the times of day to fire
 */
public record YearRepeat(int interval, YearTarget target, List<TimeOfDay> times)
    implements ScheduleExpr {
  /** Creates a new YearRepeat with defensive copy of times list. */
  public YearRepeat {
    times = List.copyOf(times);
  }
}

File: csharp/Hron/Ast/YearRepeat.cs
namespace Hron.Ast;

/// <summary>
/// A year-based repeat expression like "every year on dec 25 at 00:00".
/// </summary>
/// <param name="Interval">The number of years between occurrences (1 for every year)</param>
/// <param name="Target">The day within the year to fire</param>
/// <param name="Times">The times of day to fire</param>
public sealed record YearRepeat(int Interval, YearTarget Target, IReadOnlyList<TimeOfDay> Times) : IScheduleExpr;
